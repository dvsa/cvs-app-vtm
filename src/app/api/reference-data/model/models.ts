export * from './reference-data.model';
