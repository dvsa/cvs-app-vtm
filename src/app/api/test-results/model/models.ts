export * from './carCancelled';
export * from './carCancelledAdditionalInformation';
export * from './carCancelledAdditionalInformationLocation';
export * from './carCancelledCustomDefects';
export * from './carCancelledDefects';
export * from './carCancelledModType';
export * from './carCancelledTestTypes';
export * from './carCancelledVehicleClass';
export * from './carSubmitted';
export * from './carSubmittedAdditionalInformation';
export * from './carSubmittedDefects';
export * from './carSubmittedTestTypes';
export * from './completeTestResults';
export * from './completeTestResultsVehicleClass';
export * from './customDefect';
export * from './customDefects';
export * from './defect';
export * from './defectAdditionalInformation';
export * from './defectAdditionalInformationLocation';
export * from './defects';
export * from './hgvCancelled';
export * from './hgvSubmitted';
export * from './lgvCancelled';
export * from './lgvSubmitted';
export * from './motorcycleCancelled';
export * from './motorcycleSubmitted';
export * from './motorcycleSubmittedVehicleClass';
export * from './msUserDetails';
export * from './psvCancelled';
export * from './psvCancelledModType';
export * from './psvCancelledTestTypes';
export * from './psvSubmitted';
export * from './psvSubmittedAdditionalInformation';
export * from './psvSubmittedAdditionalInformationLocation';
export * from './psvSubmittedDefects';
export * from './psvSubmittedTestTypes';
export * from './testHistory';
export * from './testResultPutBody';
export * from './testResults';
export * from './testTypeRecords';
export * from './testTypeResults';
export * from './testTypeResultsModType';
export * from './testTypes';
export * from './trlCancelled';
export * from './trlCancelledTestTypes';
export * from './trlSubmitted';
export * from './trlSubmittedTestTypes';
