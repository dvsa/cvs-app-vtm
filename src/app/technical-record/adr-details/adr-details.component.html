<!--  <dl class="technical-record-sub-details govuk-summary-list adr-details">-->
<!--    <div class="govuk-summary-list__row">-->
<!--      <dt class="govuk-summary-list__key">-->
<!--        Able to carry dangerous goods-->
<!--        <span *ngIf="showCheck">-->
<!--          <div class="govuk-radios govuk-radios&#45;&#45;inline" id="radioADR">-->
<!--            <div class="govuk-radios__item">-->
<!--              <input class="govuk-radios__input" id="hasADR" name="hasADR" type="radio" [checked]="!isAdrNull"-->
<!--                     value="true" (change)="switchAdrDisplay($event)">-->
<!--              <label class="govuk-label govuk-radios__label" for="hasADR">-->
<!--                Yes-->
<!--              </label>-->
<!--            </div>-->
<!--            <div class="govuk-radios__item">-->
<!--              <input class="govuk-radios__input" id="hasNotADR" name="hasADR" type="radio"-->
<!--                     [checked]="isAdrNull" value="false" (change)="switchAdrDisplay($event)">-->
<!--              <label class="govuk-label govuk-radios__label" for="hasNotADR">-->
<!--                No-->
<!--              </label>-->
<!--            </div>-->
<!--          </div>-->
<!--        </span>-->
<!--      </dt>-->

<!--      <dd id="test-ableToCarry" class="govuk-summary-list__value">-->
<!--        <div *ngIf="!showCheck">-->
<!--          <span> {{ activeRecord.adrDetails == null ? "No" : "Yes"  }}</span>-->
<!--        </div>-->
<!--      </dd>-->
<!--      <dd class="govuk-summary-list__value"></dd>-->
<!--    </div>-->

<!--    <vtm-adr-details-view>-->
<!--        [activeRecord]="activeRecord"-->
<!--        [techRecordsJson]="techRecordsJson"-->
<!--        [hideForm]="hideForm">-->
<!--    </vtm-adr-details-view>-->

<!--&lt;!&ndash;    <ng-template #adrForm>&ndash;&gt;-->
<!--&lt;!&ndash;        <vtm-adr-details-form *ngIf="!hideForm"&ndash;&gt;-->
<!--&lt;!&ndash;          [vehicletypes$]="vehicletypes$"&ndash;&gt;-->
<!--&lt;!&ndash;          [permittedDangerousGoodsFe$]="permittedDangerousGoodsFe$"&ndash;&gt;-->
<!--&lt;!&ndash;          [guidanceNotesFe$]="guidanceNotesFe$"&ndash;&gt;-->
<!--&lt;!&ndash;          [initialAdrDetails]="activeRecord?.adrDetails">&ndash;&gt;-->
<!--&lt;!&ndash;        </vtm-adr-details-form>&ndash;&gt;-->
<!--&lt;!&ndash;    </ng-template>&ndash;&gt;-->

<!--  </dl>-->
