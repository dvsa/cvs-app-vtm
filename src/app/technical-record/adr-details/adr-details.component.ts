import {Component, OnInit, ViewEncapsulation, ChangeDetectionStrategy, Input} from '@angular/core';
import {Observable} from 'rxjs';
import {select, Store} from '@ngrx/store';
import {IAppState} from '@app/technical-record/adr-details/adr-details-form/store/adrDetailsForm.state';

@Component({
  selector: 'vtm-adr-details',
  templateUrl: './adr-details.component.html',
  styleUrls: ['../../app.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AdrDetailsComponent implements OnInit {

  @Input() activeRecord: any;
  @Input() showCheck: boolean;
  @Input() isAdrNull: boolean;
  @Input() adrData: boolean;
  @Input() hideForm: boolean;
  @Input() techRecordsJson: any;

  vehicletypes$: Observable<string[]>;
  permittedDangerousGoodsFe$: Observable<string[]>;
  guidanceNotesFe$: Observable<string[]>;

  constructor(private _store: Store<IAppState>) { }

  ngOnInit() {
    this.vehicletypes$ = this._store.pipe(select(s => s.vehicleTechRecordModel.vehicleTechRecordModel[0].metadata.adrDetails.vehicleDetails.typeFe));
    this.permittedDangerousGoodsFe$ = this._store.pipe(select(s => s.vehicleTechRecordModel.vehicleTechRecordModel[0].metadata.adrDetails.permittedDangerousGoodsFe));
    this.guidanceNotesFe$ = this._store.pipe(select( s => s.vehicleTechRecordModel.vehicleTechRecordModel[0].metadata.adrDetails.additionalNotes.guidanceNotesFe));
  }

  switchAdrDisplay($event): void {
    this.adrData = !($event.currentTarget.value === 'true');
    this.hideForm = $event.currentTarget.value === 'false';
  }

}
