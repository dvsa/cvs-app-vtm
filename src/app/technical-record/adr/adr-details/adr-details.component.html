<ng-container *ngIf="!edit">
  <h2 class="govuk-heading-m">ADR details</h2>

  <dl class="govuk-summary-list">
    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Vehicle type</dt>
      <dd id="test-ADR-type" class="govuk-summary-list__value">
        {{ adrDetails.vehicleDetails?.type | DefaultNullOrEmpty | titlecase }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Date processed</dt>
      <dd id="test-ADR-approvalDate" class="govuk-summary-list__value">
        {{ adrDetails.vehicleDetails?.approvalDate | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Permitted dangerous goods</dt>
      <dd id="test-ADR-permittedDangerousGoods" class="govuk-summary-list__value">
        <div *ngIf="adrDetails.permittedDangerousGoods; else hyphen">
          <div *ngFor="let perGood of adrDetails.permittedDangerousGoods; let i = index">
            <p>
              <span id="test-ADR-permittedDangerousGoods-{{ i }}">{{ perGood }}</span>
            </p>
          </div>
        </div>
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Compatibility group J</dt>
      <dd id="test-ADR-compatibilityGroupJ" class="govuk-summary-list__value">
        {{ adrDetails?.compatibilityGroupJ | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Guidance notes</dt>
      <dd id="test-ADR-additionalNotes-number" class="govuk-summary-list__value">
        <div
          *ngIf="
            this.adrDetails.additionalNotes && adrDetails.additionalNotes.number[0];
            else hyphen
          "
        >
          <div *ngFor="let number of adrDetails.additionalNotes.number; let i = index">
            <p>
              <span id="test-ADR-additionalNotes-number-{{ i }}">{{ number }}</span>
            </p>
          </div>
        </div>
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">ADR type approval number</dt>
      <dd id="test-ADR-adrTypeApprovalNo" class="govuk-summary-list__value">
        {{ adrDetails.adrTypeApprovalNo | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>
  </dl>
</ng-container>

<ng-template #hyphen><span>-</span></ng-template>

<!-- Edit Applicant Details -->
<ng-container *ngIf="edit">
  <vtm-adr-details-edit [adrDetails]="adrDetails" [metaData]="metaData"></vtm-adr-details-edit>
</ng-container>
