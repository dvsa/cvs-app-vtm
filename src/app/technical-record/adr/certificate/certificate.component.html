<ng-container *ngIf="!edit">
  <dl class="govuk-summary-list">
    <div class="govuk-summary-list__row">
      <h2 class="govuk-heading-m">Certificate</h2>
    </div>
    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Required</dt>
      <dd id="test-guidanceNotes" class="govuk-summary-list__value">
        <ng-container *ngIf="isCertificateRequested; else hyphen">
          Yes
        </ng-container>
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>
  </dl>
</ng-container>

<ng-template #hyphen><span>-</span></ng-template>

<!-- Edit Certificate required -->
<ng-container *ngIf="edit">
  <vtm-certificate-edit [hasCertificateRequested]="isCertificateRequested"></vtm-certificate-edit>
</ng-container>
