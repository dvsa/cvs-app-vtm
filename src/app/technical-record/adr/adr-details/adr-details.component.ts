import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';

import { AdrDetails } from '@app/models/adr-details';
import { MetaData } from '@app/models/meta-data';

@Component({
  selector: 'vtm-adr-new-details',
  templateUrl: './adr-details.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AdrDetailsComponent implements OnInit {
  @Input() edit: boolean;
  @Input() adrDetails: AdrDetails;
  @Input() metaData: MetaData;

  constructor() {}

  ngOnInit() {}
}
