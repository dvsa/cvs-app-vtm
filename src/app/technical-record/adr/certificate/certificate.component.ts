import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';

import { AdrDetails } from '@app/models/adr-details';
import { NOTES } from '@app/app.enums';

@Component({
  selector: 'vtm-certificate',
  templateUrl: './certificate.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class CertificateComponent implements OnInit {
  @Input() edit: boolean;
  @Input() adrDetails: AdrDetails;

  isCertificateRequested: boolean;

  constructor() {}

  ngOnInit() {
    this.isCertificateRequested = !!this.getCertificateRequested(
      this.adrDetails.additionalNotes.guidanceNotes
    );
  }

  getCertificateRequested(notes: string[]) {
    return notes.find((value) => NOTES.GUIDANCENOTE_CODE === this.mapGuidanceNotesToCodes(value));
  }

  mapGuidanceNotesToCodes(value: string): string {
    if (value) {
      return value
        .toLowerCase()
        .trim()
        .replace(/ /gi, '');
    }
  }
}
