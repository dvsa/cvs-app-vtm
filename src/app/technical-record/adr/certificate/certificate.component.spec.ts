import { AdditionalNotes } from './../../../models/adr-details';
import { CUSTOM_ELEMENTS_SCHEMA, Component, Input } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { SharedModule } from '@app/shared';
import { TESTING_UTILS } from '@app/utils/testing.utils';
import { AdrDetails } from '@app/models/adr-details';
import { CertificateComponent } from './certificate.component';

const adrWithCertRequested = TESTING_UTILS.mockAdrDetails({
  additionalNotes: {
    guidanceNotes: ['New certificate requested']
  } as AdditionalNotes
}) as AdrDetails;

describe('CertificateComponent', () => {
  let component: CertificateComponent;
  let fixture: ComponentFixture<CertificateComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [CertificateComponent, TestCertificateEditComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CertificateComponent);
    component = fixture.componentInstance;
    component.edit = false;
    component.adrDetails = TESTING_UTILS.mockAdrDetails();
  });

  it('should create view only with populated data', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should set certificate requested to truthy', () => {
    component.adrDetails = adrWithCertRequested;
    fixture.detectChanges();

    expect(component.isCertificateRequested).toBeTruthy();
  });

  it('should render the editable component if edit is true', () => {
    component.adrDetails = adrWithCertRequested;
    component.edit = true;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});

@Component({
  selector: 'vtm-certificate-edit',
  template: `
    <div>Value of Certificate Request is {{ hasCertificateRequested }}</div>
  `
})
class TestCertificateEditComponent {
  @Input() hasCertificateRequested: boolean;
}
