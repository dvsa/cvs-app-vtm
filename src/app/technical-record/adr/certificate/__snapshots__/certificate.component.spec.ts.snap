// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CertificateComponent should create view only with populated data 1`] = `
<vtm-certificate
  adrDetails={[Function Object]}
  edit="false"
  isCertificateRequested="false"
>
  <dl
    class="govuk-summary-list"
  >
    <div
      class="govuk-summary-list__row"
    >
      <h2
        class="govuk-heading-m"
      >
        Certificate
      </h2>
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Required
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-guidanceNotes"
      >
        
        <span>
          -
        </span>
      </dd>
      <dd
        class="govuk-summary-list__value"
      />
    </div>
  </dl>
</vtm-certificate>
`;

exports[`CertificateComponent should render the editable component if edit is true 1`] = `
<vtm-certificate
  adrDetails={[Function Object]}
  edit={[Function Boolean]}
  isCertificateRequested={[Function Boolean]}
>
  <vtm-certificate-edit
    ng-reflect-has-certificate-requested="true"
  >
    <div>
      Value of Certificate Request is true
    </div>
  </vtm-certificate-edit>
</vtm-certificate>
`;
